common/app/views/main.scala.html

```diff
@@ -27,34 +28,110 @@
     }
 }
 
-@immersiveMainMedia(page: GalleryPage) = {
-    <div class="new-gallery content tonal--tone-media immersive-main-media">
-        @defining(
-            page.item.content.trail.trailPicture.head
-        ) { picture =>
-            @fragments.image(
-                picture = picture,
-                classes = Seq("immersive-main-media__img"),
-                widths = MainMedia.immersive,
-                imageAltText = "",
-                isImmersiveMainMedia = true
-            )
+@immersiveMainMedia(page: ContentPage) = {
+    <div class="content content--immersive tonal tonal--@toneClass(page.item) immersive-main-media
+        @if(page.item.tags.isGallery) { new-gallery }
+        @if(page.item.tags.isArticle) { content--immersive-article }
+    ">
+        @if(!page.item.tags.isGallery) {
+            <div class="gs-container immersive-main-media__logo">
+                <a href="@LinkTo {/}">
+                    <span class="u-h">The Guardian</span>
+                    @fragments.inlineSvg("guardian-logo-160", "logo", Seq("immersive-main-media__logo__svg"))
+                </a>
+                @page.item match {
+                    case article: Article => {
+                        @if(article.isUSMinute) {
+                            <a class="logo--us-election" href="@LinkTo {/us-news/us-elections-2016}">
+                                <span class="u-h">The US Election 2016</span>
+                                @fragments.inlineSvg("us-election-logo", "logo")
+                            </a>
+                        }
+                    }
+                    case _ => {}
+                }
+            </div>
+        }
+
+        @if(page.item.tags.isArticle && (page.item.elements.hasMainEmbed || page.item.elements.elements("main").isEmpty)) {
+            <div class="immersive-main-media__media">
+                <div class="immersive-main-media__media__loading">
+                    <div class="immersive-main-media__media__loading__animation is-updating"></div>
+                    <span class="u-h">Loading header</span>
+                </div>
+                @Html(page.item.fields.main)
+            </div>
+        } else {
+            @page.item.elements.mainPicture.map(_.images).orElse(page.item.trail.trailPicture).map { picture =>
```

Comment:
Yes! Perhaps we could have a `immersiveMedia` option on the `ContentPage` type? What do you think to that?